backend/backendutil/search.go

```diff
@@ -41,16 +42,28 @@ type lengther interface {
 	Len() int
 }
 
+type countWriter struct {
+	N int
+}
+
+func (w *countWriter) Write(b []byte) (int, error) {
+	w.N += len(b)
+	return len(b), nil
+}
+
 func bodyLen(e *message.Entity) (int, error) {
+	headerSize := countWriter{}
+	textproto.WriteHeader(&headerSize, e.Header.Header)
+
 	if l, ok := e.Body.(lengther); ok {
-		return l.Len(), nil
+		return l.Len() + headerSize.N + 2 /* CRLF for empty line */, nil
```

Comment:
Doesn't `WriteHeader` already write the final double CRLF?

Reply:
Hm, gotcha, you are right.